topdown/cache/cache.go

```diff
@@ -154,16 +154,14 @@ func NewInterQueryCache(config *Config) InterQueryCache {
 func NewInterQueryCacheWithContext(ctx context.Context, config *Config) InterQueryCache {
 	iqCache := newCache(config)
 	if iqCache.staleEntryEvictionTimePeriodSeconds() > 0 {
-		cleanupTicker := time.NewTicker(time.Duration(iqCache.staleEntryEvictionTimePeriodSeconds()) * time.Second)
 		go func() {
+			cleanupTicker := time.NewTicker(time.Duration(iqCache.staleEntryEvictionTimePeriodSeconds()) * time.Second)
+			defer cleanupTicker.Stop()
 			for {
 				select {
 				case <-cleanupTicker.C:
-					cleanupTicker.Stop()
 					iqCache.cleanStaleValues()
-					cleanupTicker = time.NewTicker(time.Duration(iqCache.staleEntryEvictionTimePeriodSeconds()) * time.Second)
```

Comment:
Are we ok with this in scenarios where `cleanStaleValues` routine can consume more time than `staleEntryEvictionTimePeriodSeconds`?

Reply:
Okay, I've returned this to the pattern before, with a comment.  This seems like we're already on the pathological side of behavior, because for a 5s eviction ticker to run "continuously", you'd need to have paused the cache entirely for 5s -- with the original code I've re-added, you'd have a 5s pause, then 5s of checks, then 5s pause, etc.  That would be pretty unpleasant for most interactive use, though a bulk tool might manage.